assets/javascripts/filter_form.js

```diff
@@ -47,10 +48,13 @@ function parseFilterArguments(paramHandler) {
         input.attr('value', val);
         input.attr('name', key);
         input.attr('hidden', true);
-        $('#filter-form').append(input);
+        hiddenInputs.push(input);
       }
     }
   }
+  for (var i = 0; i < hiddenInputs.length; i++) {
+    $('#filter-form').append(hiddenInputs[i]);
+  }
```

Comment:
I would avoid the repeated calls to `$('#filter-form')` in the loop and stick to standard JavaScript (as we probably shouldn't introduce more jQuery code at this point):
```suggestion
  const filterForm = document.getElementById('filter-form');
  hiddenInputs.forEach(i => filterForm.appendChild(i));
```

Probably the code that creates the input elements should also just use `document.createElement('input')` and normal DOM APIs.

It is not obvious at all why your change makes a difference but it is because of what the code passed via `paramHandler` does. That means you could probably also just change *that* code and make it skip those elements, e.g. replace `if (formElement) {` with `if (formElement && !formElement.hidden) {`. This would avoid an extra array and loop.

Reply:
I tried now:
```
if (formElement && !formElement.hidden)
```
but then it is always skipping the last distri mentioned in the query parameters...
I need to debug this.

Reply:
So apparently in the first call of the `paramHandler` `form[key]` is undefined. In the second it''s returning `HTMLInputElement` (hidden). After that it returns a `RadioNodeList`. Possibly because there are now two elements already. Not sure yet how I should handle that.

Reply:
Also I would like to add that I didn't introduce any new jquery code. I just mvoed it from one loop to the other.

Reply:
Ok, then your approach might be simpler after all. And it is true that the code was just moved.